### DISCUSSION TITLE
Feature request: Popover.onClose()

### DISCUSSION THREAD

**Discussion by vnugent** [tailwindlabs/headlessui#1302] [Ideas]
When users close the popover by clicking outside, pressing ESC, or clicking on a custom Cancel button, I want to perform some action, for example, reset form states to their default values.  

[`Dialog`](https://headlessui.dev/react/dialog#dialog) already supports `onClose()`.  Can you add it to PopOver as well?  Thanks!

**Comment by jarrodmjack** [comment]
did this ever get added?

**Comment by nick-tw** [comment]
It's crazy that Popover doesn't have this feature. Also does not have logic where popover might be bleeding over the left or right edges of the window and needs to handle that scenario. Unfortunately have to look for another project and replace all of my headless ui elements.

**Reply by vnugent** [reply]
In the end I migrated to Radix UI.

**Comment by matijaSos** [comment]
Also very surprised this isn't supported. Is there any particular reason for it? 

BTW why nobody from the maintainers ever answered to this discussion? Should we move it somewhere else, under issues?

**Comment by mnezmah** [comment]
any workarounds?

**Reply by mnezmah** [reply]
I've found it here: https://github.com/tailwindlabs/headlessui/discussions/1901
